### PR TITLE
Pcp api

### PR DESCRIPTION
**Pull request by alar77**
Get, Create, Delete method added
Test included

### CODE REVIEW

File: api/v3/Pcp.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * Create or update a survey.
+ *
+ * @param array $params
+ *          Array per getfields metadata.
+ *          
+ * @return array api result array
+ */
+function civicrm_api3_pcp_create($params) {
+  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), 
+      $params, 'Pcp');
```
Comment (seamuslee001) [inline api/v3/Pcp.php]: @ggargani Not sure why you need 'Pcp' here and in the 3 relevant function below, you should be able to get away with $params on the one line with the rest, looks like i think your editor is tied to the 80char limit

File: tests/phpunit/api/v3/PcpTest.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+class api_v3_PcpTest extends CiviUnitTestCase {
+  protected $params;
+  protected $entity = 'Pcp';
+  public $DBResetRequired = TRUE;
+
+  public function setUp() {
+    $this->params = array ('title' => "Pcp title", 
+        // 'status_id' => 0,
+        'contact_id' => 1, 'page_id' => 1, 'pcp_block_id' => 1);
+    /*
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/PcpTest.php]: This should be removed, no need for debugging stuff in core

**Comment by seamuslee001** [comment]
@ggargani I have just run the civilint from buildkit on your new files they report

FILE: .../buildkit/build/47.demo/sites/all/modules/civicrm/api/v3/Pcp.php
----------------------------------------------------------------------
FOUND 7 ERRORS AFFECTING 7 LINES
----------------------------------------------------------------------
 40 | ERROR | [x] Whitespace found at end of line
 44 | ERROR | [x] Whitespace found at end of line
 68 | ERROR | [x] Whitespace found at end of line
 72 | ERROR | [x] Whitespace found at end of line
 83 | ERROR | [x] Whitespace found at end of line
 87 | ERROR | [x] Whitespace found at end of line
 89 | ERROR | [x] Expected 1 newline at end of file; 0 found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 7 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

FILE: ....demo/sites/all/modules/civicrm/tests/phpunit/api/v3/PcpTest.php
----------------------------------------------------------------------
FOUND 21 ERRORS AFFECTING 16 LINES
----------------------------------------------------------------------
  45 | ERROR | [x] Whitespace found at end of line
  46 | ERROR | [x] Expected 1 space after asterisk; 9 found
  54 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
  54 | ERROR | [x] The first index in a multi-value array must be on
     |       |     a new line
  54 | ERROR | [x] Whitespace found at end of line
  56 | ERROR | [x] Array indentation error, expected 6 spaces but
     |       |     found 8
  56 | ERROR | [x] Each index in a multi-line array must be on a new
     |       |     line
  56 | ERROR | [x] Each index in a multi-line array must be on a new
     |       |     line
  74 | ERROR | [x] Whitespace found at end of line
  88 | ERROR | [x] Whitespace found at end of line
  99 | ERROR | [x] Whitespace found at end of line
 100 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
 102 | ERROR | [x] Whitespace found at end of line
 103 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
 104 | ERROR | [x] Whitespace found at end of line
 105 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
 119 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
 121 | ERROR | [x] Whitespace found at end of line
 123 | ERROR | [x] There must be no space between the "array" keyword
     |       |     and the opening parenthesis
 125 | ERROR | [x] Expected 1 newline at end of file; 0 found
 125 | ERROR | [x] The closing brace for the class must have an empty
     |       |     line before it
----------------------------------------------------------------------
PHPCBF CAN FIX THE 21 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Can you fix up those style issues

File: api/v3/Pcp.php (hunk 1 above)
Comment (seamuslee001) [inline api/v3/Pcp.php]: @ggargani I tested locally changing this to ```php 

return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
``` 
worked fine when i ran your unit test

**Comment by seamuslee001** [comment]
@totten @monishdeb Can either of you please ok to test this?

**Comment by totten** [comment]
jenkins, add to whitelist

**Comment by ggargani** [comment]
@seamuslee001  Wow, you were superfast :) I was still getting used with the release process (and overlooked the autoformatting from eclipse).
I was tunning civilint while you were commenting, sorry for your time waste.

> vagrant@civi:~/civi47$ civilint api/v3/Pcp.php  tests/phpunit/api/v3/PcpTest.php 
> ===========================[ Identify Files ]===========================
> PHP Files:
>  * api/v3/Pcp.php
>  * tests/phpunit/api/v3/PcpTest.php
> ===============================[ php -l ]===============================
> ===============================[ phpcs  ]===============================

Also

> vagrant@civi:~/civi47/tools$ php ./scripts/phpunit --color=always api_v3_PcpTest
> PHPUnit 4.8.21 by Sebastian Bergmann and contributors.
> 
> .
> Installing d7masterte_xl07w database
> ...
> 
> Time: 18.47 seconds, Memory: 40.00Mb
> 
> OK (4 tests, 28 assertions)
>

**Comment by seamuslee001** [comment]
nice :+1: We'll let the full civisuite run but i think this is pretty good :-)

File: api/v3/Pcp.php (hunk 1 above)
Comment (ggargani) [inline api/v3/Pcp.php]: Well, I saw a couple of api and oticed that some of them uses the third parameter other ones not, so I was not sure about the best practise

**Comment by seamuslee001** [comment]
@eileenmcnaughton @totten @monishdeb Test failure is unrelated think we can merge this

**Comment by eileenmcnaughton** [comment]
Wow huge effort guys - so many people have started on this api but somehow never completed it but finally you got it all the way!!

**Pull request merged**